presto-main/src/main/java/com/facebook/presto/sql/planner/PlanPrinter.java

```diff
@@ -418,38 +481,73 @@ private void printStats(int indent, PlanNodeId planNodeId, boolean printInput, b
             return;
         }
 
-        double fraction = (nodeStats.getWallTime().toMillis()) / (double) totalMillis;
-        String fractionString;
-        if (isFinite(fraction)) {
-            fractionString = format(Locale.US, "%.2f%%", 100.0 * fraction);
-        }
-        else {
-            fractionString = "?";
-        }
+        double fraction = 100.0d * nodeStats.getPlanNodeWallTime().toMillis() / totalMillis;
 
         output.append(indentString(indent));
-        output.append("Cost: " + fractionString);
+        output.append("Cost: " + formatDouble(fraction) + "%");
         if (printInput) {
             output.append(format(", Input: %s (%s)",
-                    formatPositions(nodeStats.getInputPositions()),
-                    nodeStats.getInputDataSize().toString()));
+                    formatPositions(nodeStats.getPlanNodeInputPositions()),
+                    nodeStats.getPlanNodeInputDataSize().toString()));
         }
         output.append(format(", Output: %s (%s)",
-                formatPositions(nodeStats.getOutputPositions()),
-                nodeStats.getOutputDataSize().toString()));
+                formatPositions(nodeStats.getPlanNodeOutputPositions()),
+                nodeStats.getPlanNodeOutputDataSize().toString()));
         if (printFiltered) {
-            double filtered = 100.0 * (nodeStats.getInputPositions() - nodeStats.getOutputPositions()) / nodeStats.getInputPositions();
-            String filteredString;
-            if (isFinite(filtered)) {
-                filteredString = format(Locale.US, "%.2f%%", filtered);
-            }
-            else {
-                filteredString = "?";
+            double filtered = 100.0d * (nodeStats.getPlanNodeInputPositions() - nodeStats.getPlanNodeOutputPositions()) / nodeStats.getPlanNodeInputPositions();
+            output.append(", Filtered: " + formatDouble(filtered) + "%");
+        }
+        output.append('\n');
+
+        printDistributions(indent, nodeStats);
+    }
+
+    private void printDistributions(int indent, PlanNodeStats nodeStats)
+    {
+        Map<String, Long> totalDrivers = nodeStats.getOperatorTotalDrivers();
```

Comment:
For now I don't think we should be printing information about operators. Is there a case where this adds a lot of value? For the most part they map 1:1 to plan nodes, and in the future that should be even more true